department-of-veterans-affairs/vets-website
# VACMS-18536 Remove duplicate analytics events in Facility Locator

## Are you removing, renaming or moving a folder in this PR?

## Summary
Remove duplicate Google Analytics events in Facility locator. These were less obvious than usual; in many cases we had code that was running multiple times and shouldn't have been, and thus logging more events.

## Related issue(s)
- https://github.com/department-of-veterans-affairs/va.gov-cms/issues/18536

## Testing done
I used the Chrome Adswerve plugin to test the below scenarios:

<details><summary>Searching for facilities</summary>

1. Go to `/find-locations`
2. Enter any search terms
3. Click "Search"
4. Validate that you see the `fl-search-results` event **only once**

<img width="500" alt="Screenshot 2024-08-08 at 1 11 50 PM" src="https://github.com/user-attachments/assets/c5d45706-adbf-4e6a-90b6-2592325f62b4">
</details>

<details><summary>Moving the map with facility results</summary>

1. Go to `/find-locations`
2. Enter any search terms
3. Click "Search"
4. Click into the map and move the map view
5. Validate that you see the `fl-search` event once with `fl-map-miles-moved`
6. Validate that you do not see the `fl-search-results` event at all
7. If the "Search this area of the map" button appears when you move the map, click it
8. Validate that you see the `fl-search` event **only once**
9. If there are search results from that new search, validate that you see the `fl-search-results` event **only once**
</details>

<details><summary>Zooming the map with facility results</summary>

1. Go to `/find-locations`
2. Enter any search terms
3. Click "Search"
4. Hover over the map (click to focus on it if you need to) and use your scroll wheel / track pad to zoom in or out
5. Validate that you see `fl-map-zoom-in` or `fl-map-zoom-out` once per zoom interaction as appropriate (this interaction can be flaky and is the same way in prod. Sometimes you have to zoom in/out a couple of times for the event to register).

<img width="600" alt="Screenshot 2024-08-08 at 4 06 11 PM" src="https://github.com/user-attachments/assets/180f3b13-e47f-44bb-8c75-82d96d36e89f">
<img width="600" alt="Screenshot 2024-08-08 at 4 06 07 PM" src="https://github.com/user-attachments/assets/8058b0e4-a16d-4c91-9eee-d0987a59d726">


</details>

<details><summary>Clicking on a pin on the map with facility results</summary>

1. Go to `/find-locations`
2. Enter any search terms
3. Click "Search"
4. Click on a pin on the map (gray circle with a white letter in it)
5. Validate that you see the `fl-map-pin-click` once per pin click

<img width="600" alt="Screenshot 2024-08-08 at 4 08 12 PM" src="https://github.com/user-attachments/assets/7a344b1f-f403-4ea1-aad5-626d7343d09a">


</details>

<details><summary>Clicking facility results</summary>

1. Go to `/find-locations`
2. Enter any search terms
3. Click "Search"
4. Click one of the search results
10. Validate that you see the `fl-results-click` like the below **only once**

<img width="500" alt="Screenshot 2024-08-08 at 1 12 40 PM" src="https://github.com/user-attachments/assets/0301790c-f80b-4764-9a74-9ba11907402e">

</details>

<details><summary>Using keyboard to select facility results</summary>

1. Go to `/find-locations`
2. Enter any search terms
3. Click "Search"
4. Tab to one of the search results, then press the enter key when the link is focused
5. Validate that you see the `fl-results-click` like the below **only once**

<img width="500" alt="Screenshot 2024-08-08 at 1 12 40 PM" src="https://github.com/user-attachments/assets/0301790c-f80b-4764-9a74-9ba11907402e">

</details>

<details><summary>Searching for urgent care facilities</summary>

1. Go to `/find-locations`
2. Enter the below search criteria
<img width="600" alt="Screenshot 2024-08-08 at 3 12 31 PM" src="https://github.com/user-attachments/assets/0c98c19c-2470-4e16-bf19-0c27ca248dee">

3. Click on this alert link on one of the results
<img width="300" alt="Screenshot 2024-08-08 at 3 12 34 PM" src="https://github.com/user-attachments/assets/79a6fa2a-06d7-4821-bfbe-b833ef1e2f38">

4. Validate that you see the below event
<img width="400" alt="Screenshot 2024-08-08 at 3 06 46 PM" src="https://github.com/user-attachments/assets/a5ba8eef-0855-4410-96d2-29b927807fd9">

5. Validate that you do **not** see the `cta-primary-button-click` event

</details>

<details><summary>Searching for emergency care facilities</summary>


1. Go to `/find-locations`
2. Enter the below search criteria
<img width="600" alt="Screenshot 2024-08-08 at 3 21 44 PM" src="https://github.com/user-attachments/assets/c8c3bcc8-b580-43bf-a1b9-18988242bb84">

3. Click on this alert link on one of the results
<img width="295" alt="Screenshot 2024-08-08 at 3 22 50 PM" src="https://github.com/user-attachments/assets/0b474d05-452f-48e0-afb3-152e4a7f126e">

4. Validate that you see the below event
<img width="600" alt="Screenshot 2024-08-08 at 3 21 44 PM" src="https://github.com/user-attachments/assets/9d41d4cf-a595-4d9d-a9ee-160262d70d64">

5. Validate that you do **not** see the `cta-emergency-benefit-button-click` event

</details>

## Screenshots
See `Testing done` for all screenshots.

## What areas of the site does it impact?

Facility locator only.

## Acceptance criteria
- [x] Remove duplicate Google Analytics events where possible
